From f3e2d3999be2889a559423d03c5bbbbd32d4b8c6 Mon Sep 17 00:00:00 2001
Date: Wed, 29 May 2019 19:10:31 +0300
Subject: [PATCH] Fix setgroups declaration

In POSIX, 'ngroups' parameter of 'setgroups' function is of type
'size_t' instead of 'int'.

Signed-off-by: Costin Lupu <costin.lupu@cs.pub.ro>
---
 newlib/libc/include/sys/unistd.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/newlib/libc/include/sys/unistd.h b/newlib/libc/include/sys/unistd.h
index 75f8a51..bcb2f69 100644
--- a/newlib/libc/include/sys/unistd.h
+++ b/newlib/libc/include/sys/unistd.h
@@ -188,7 +188,7 @@ int     _EXFUN(seteuid, (uid_t __uid ));
 int     _EXFUN(setgid, (gid_t __gid ));
 #endif
 #if __BSD_VISIBLE
-int	_EXFUN(setgroups, (int ngroups, const gid_t *grouplist ));
+int	_EXFUN(setgroups, (size_t ngroups, const gid_t *grouplist ));
 #endif
 #if __BSD_VISIBLE || (__XSI_VISIBLE && __XSI_VISIBLE < 500)
 int	_EXFUN(sethostname, (const char *, size_t));
-- 
2.11.0

